lib/tasks/dev.rake

```diff
@@ -0,0 +1,94 @@
+desc "Fill the database tables with some sample data"
+task({ :sample_data => :environment}) do
+  
+  User.destroy_all
+  Loan.destroy_all
+  Adjustment.destroy_all
+
+  # Create Sample Users
+    #  email                  :string           default(""), not null
+    #  encrypted_password     :string           default(""), not null
+
+  user_hashes = [
+    { email: "alice@example.com", password: "password" },
+    { email: "bob@example.com", password: "password" },
+    { email: "carol@example.com", password: "password" }
+  ]
+
+  User.create(user_hashes)
+
+  p "Added #{User.count} Users"
+
+  # Create Sample Loans
+    #  current_balance     :float
+    #  original_amount     :float
+    #  interest_rate       :float
+    #  periods_in_year     :integer
+    #  user_id             :integer
+    #  loan_name           :string
+    #  monthly_min_payment :float
+
+    #cur_bal = [1000, 5000, 10000, 15000, 20000, 25000, 30000, 35000, 40000]
+  EXAMPLE_ORIGINAL_BALANCES = [1000, 5000, 10000, 15000, 20000, 25000, 30000, 35000, 40000]
+  EXAMPLE_INTEREST_RATES = [2, 3, 4, 5, 6]
+  EXAMPLE_PAYBACK_IN_YEARS = [2, 4, 6]
+  EXAMPLE_PERIODS_IN_A_YEAR  = 12
+  users = User.all
+  EXAMPLE_LOAN_NAMES = ["Car", "Car 2", "Student Loan", "Grad School", "Credit Card"]
+
+  users.each do |user|
+    3.times do
+      loan = Loan.new
+      loan.current_balance = EXAMPLE_ORIGINAL_BALANCES.sample
+      loan.original_amount = loan.current_balance
+      loan.interest_rate = EXAMPLE_INTEREST_RATES.sample
+      loan.periods_in_year = EXAMPLE_PERIODS_IN_A_YEAR
+      loan.user = user
```

Comment:
Since a User `has_many` Loans, you're allowed to do

```rb
loan = user.loans.new
```
which is equivalent to
```rb
loan = Loan.new
loan.user = user
```
---

You could also use `create` and wrap all the column assignments in parentheses

```rb
loan = user.loans.create(
  current_balance: EXAMPLE_ORIGINAL_BALANCES.sample,
  interest_rate: EXAMPLE_INTEREST_RATES.sample,
  ...
)
then you can remove the `.save`
```

Reply:
Thanks! Really good reminders about all the helper methods and how much they can clean up and simplify code.  I also need to remember that `create` is an option as well.  Appreciate the insights!